pallets/farming/src/lib.rs

```diff
@@ -76,7 +79,18 @@ pub mod pallet {
 	pub trait Config: frame_system::Config {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 
-		type MultiCurrency: MultiCurrency<AccountIdOf<Self>, CurrencyId = CurrencyId>;
+		type CurrencyId: FullCodec
```

Comment:
Is this really required? Why not simply use `CurrencyIdOf<T>` instead of `T::CurrencyId`. This is required anyway due to how the storage `PoolInfos` is defined: it returns a value of type `PoolInfo<BalanceOf<T>, CurrencyIdOf<T>, AccountIdOf<T>, BlockNumberFor<T>>` and then at some other part of the code it is expected anyway that `CurrencyIdOf<T>` and `T::CurrencyId` are equal.

Reply:
Ah, I see that you also changed the definition of the type `type MultiCurrency`. Okay, then I will accept.